packages/opentelemetry-api/src/metrics/Metric.ts

```diff
@@ -77,7 +78,14 @@ export enum ValueType {
  * Metric represents a base class for different types of metric
  * pre aggregations.
  */
-export interface Metric<T> {
+export interface Metric {
+  /**
+   * Clears all timeseries from the Metric.
+   */
+  clear(): void;
+}
+
+export interface UnboundMetric<T> extends Metric {
```

Comment:
I think we can rename the generic type `T` to something like `Bound` and assert that it extends `Metric` right? Otherwise `bind` can actually return anything and the types will not complain.

Reply:
The `T` in `UnboundMetric<T>` doesn't stand for `Metric` but `BoundInstrument`. 🤔 Although `UnboundMetric<T>` doesn't restrict the type parameter, concrete metric types like Counter do restrict it, like `interface Counter extends UnboundMetric<BoundCounter>`, in which `BoundCounter` is what the `bind` returns.

Reply:
It is true `UnboundMetric<T>.bind` can return arbitrary types if the `T` overrides with arbitrary types. In the PR there are `Counter`, `Measure`, and `Observer` exported to replace `UnboundMetric<T>` in most use cases.